perceval/components/linear_circuit.py

```diff
@@ -581,11 +549,11 @@ def decomposition(U: MatrixN,
                 assert isinstance(constraint, (list, tuple)) and len(constraint) == len(component.get_parameters()),\
                     "there should as many component in each constraint than free parameters in the component"
         while count < max_try:
-            if shape == "triangle":
+            if shape == InterferometerShape.TRIANGLE:
                 lc = decomposition.decompose_triangle(U, component, phase_shifter_fn, permutation, precision,
                                                       constraints, allow_error=allow_error,
                                                       ignore_identity_block=ignore_identity_block)
-            else:
+            elif shape == InterferometerShape.RECTANGLE:
                 lc = decomposition.decompose_rectangle(U, component, phase_shifter_fn, permutation, precision,
                                                        constraints, allow_error=allow_error,
                                                        ignore_identity_block=ignore_identity_block)
```

Comment:
consider handle the else case (with a not implemented exception for exemple) if you plan one day of enriched the InterferometerShape enum

Reply:
Done